### PR TITLE
fix(firefox): throw error when added script blocked by CSP

### PR DESCRIPTION
**Pull request by yury-s**

### CODE REVIEW

File: src/frames.ts [hunk 1]
```diff
@@ -566,12 +566,19 @@ export class Frame {
...
         contents += '//# sourceURL=' + path.replace(/\n/g, '');
-        return (await context.evaluateHandleInternal(addScriptContent, { content: contents, type })).asElement()!;
+        result = (await context.evaluateHandleInternal(addScriptContent, { content: contents, type })).asElement()!;
+      } else {
+        result = (await context.evaluateHandleInternal(addScriptContent, { content: content!, type })).asElement()!;
       }
-      return (await context.evaluateHandleInternal(addScriptContent, { content: content!, type })).asElement()!;
+      // Another round trip to the browser to ensure that we receive CSP error messages
+      // (if any) logged asynchronously in a separate task on the content main thread.
+      if (this._page._delegate.cspErrorsAsynchronousForInlineScipts)
```
Comment (dgozman) [inline src/frames.ts:579]: We should probably do the same for inline styles and test it.

File: src/frames.ts [hunk 2]
```diff
@@ -566,12 +566,19 @@ export class Frame {
...
+        await context.evaluateInternal(() => true);
```
Comment (dgozman) [inline src/frames.ts:580]: Have you found the source of this? Are we sure it is a single task?

**Review by dgozman — APPROVED** [review]

File: src/frames.ts (hunk 1 above)
Comment (yury-s) [reply src/frames.ts:579]: No, styles are different. I thought the same but the styles seem to be handled asynchronously and they support onload/error events. We already have a [test](https://github.com/microsoft/playwright/blob/master/test/page.spec.js#L725) for this.

File: src/frames.ts (hunk 2 above)
Comment (yury-s) [reply src/frames.ts:580]: The console logging [happens](http://ffsearch.azurewebsites.net/#path=%2Fhome%2Fjoe%2Ffirefox%2Fdom%2Fsecurity%2FnsCSPContext.cpp&line=1512) in a _separate_ task on the main thread. My assumption here is that the second evaluate is handled in its own task after the logging, so there is still a room for false positives. I could add some hooks in the native part to report the error synchronously via juggler but I felt like the problem we are fixing doesn't warrant it.

**Pull request merged**